### ISSUE TITLE
[BUG] Temperature applied twice in FlashInfer sampling path

### ISSUE THREAD

**Issue by KJLdefeated** [Flink-ddd/RL-Engine#81]
Thanks author for interesting project! But I found some bug(?) in SamplerBackend.

SamplerBackend.sample() divides logits by temperature once for all backends, then the FlashInfer branch divides by temperature a second time.

## Location
[rl_engine/kernels/sampling.py:42-68](https://github.com/RL-Align/RL-Kernel/blob/main/rl_engine/kernels/sampling.py#L42-L68)

```
def sample(self, logits, top_k=None, top_p=None, temperature=1.0, deterministic=True):
    logits = logits.contiguous()
    if temperature != 1.0:
        logits = logits / temperature          # division #1 (all backends)

    if self.backend == constants.BackendLib.FLASHINFER.value:
        ...
        logits = logits.float().contiguous()
        if temperature != 1.0:
            logits.div_(temperature)            # division #2 (FlashInfer only) ← BUG
        probs = torch.softmax(logits, dim=-1)
```

I can make a PR to fix it.

**Comment by Flink-ddd** [comment]
Thank you for wanting to fix this issue.
